mesonbuild/interpreter/interpreter.py

```diff
@@ -72,6 +70,36 @@
                             build.BuildTarget, build.CustomTargetIndex,
                             build.GeneratedList]
 
+    class AddProjectArgsKwArgs(TypedDict):
```

Comment:
I don't like the name of this class... I know why you want the `KwArgs` suffix but still. I don't have a better suggestion though, so...

Also, this has a different naming convention compared to `FuncGeneratorKwArgs` from your QT PR.

Reply:
You're right, it is. I like the KwArgs because it makes it clear that this is the Keyword Arguments for a function, but I could be convinced to use a different convention

Reply:
I am not sure if this the best idea, but we could make a new `typing` package in `interpreter` with a `kwargs` module. Then we could use `kwargs.AddProjectArgs`. This would also have the benefit of reducing the size of the `interpreter.py`, which is still too large for my taste.

Reply:
I like that idea, I'll do that.